src/error.rs

```diff
@@ -0,0 +1,15 @@
+use std::fmt;
+use std::fmt::Formatter;
+
+#[derive(Debug)]
+pub enum IdenticonError {
+    SaveImageError,
+}
+
+impl fmt::Display for IdenticonError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        match self {
+            IdenticonError::SaveImageError => write!(f, "could not save image"),
```

Comment:
With this the indented formater `"{:#?}"` keeps working
```suggestion
            IdenticonError::SaveImageError => f.write_str("could not save image"),
```

Reply:
I don't doubt this, but do you have documentation or a blog post or the like I can read about this? I'd love to learn more and this seems like an odd oversight of the `write!` macro if that is the case

Reply:
There is the documentation of the [std:.fmt::Formatter](https://doc.rust-lang.org/std/fmt/struct.Formatter.html) but I don't see any good application for implementing the recursion for `Display`. The standard library is geared towards `Debug`.
I would like to investigate this some more, maybe write a blog post ^^

After some digging I found this main difference:
```rust
impl std::fmt::Display for Displayable {
    fn fmt(&self, f: &mut std::fmt::Formatter) -> Result<(), std::fmt::Error> {
        match self {
            Displayable::Write =>  write!(f, "'simple write'"),
            Displayable::Formatter if f.alternate() => f.write_str("^formatted alternative^"),
            Displayable::Formatter => f.write_str("^formatted write^"),
        }
    }
}
```
This desugares into:
```rust
impl std::fmt::Display for Displayable {
    fn fmt(&self, f: &mut std::fmt::Formatter) -> Result<(), std::fmt::Error> {
        match self {
            Displayable::Write => f.write_fmt(::core::fmt::Arguments::new_v1(
                &["\'simple write\'"],
                &match () {
                    () => [],
                },
            )),
            Displayable::Formatter if f.alternate() => f.write_str("^formatted alternative^"),
            Displayable::Formatter => f.write_str("^formatted write^"),
        }
    }
}
```

I assumed that the compiler could optimize it to more or less the same in release mode.
However `write_fmt(...)` call is still around in the assembler.
So, it is a little faster

Reply:
Thank you! This is oddly fascinating, especially given most examples use the `write!` macro. I'll switch over to your suggestion but I'll drop a comment in case this presents any issues in the future.